src/metabase/integrations/ldap/default_implementation.clj

```diff
@@ -33,23 +32,34 @@
              :size-limit 1}))
           u/lower-case-map-keys))
 
+(s/defn ^:private process-group-membership-filter :- su/NonBlankString
+  "Replace DN and UID placeholders with values returned by the LDAP server."
+  [group-membership-filter :- su/NonBlankString
+   dn                      :- su/NonBlankString
+   uid                     :- (s/maybe su/NonBlankString)]
+  (let [uid-string (or uid "")]
+    (-> group-membership-filter
+        (str/replace "{dn}" (Filter/encodeValue ^String dn))
+        (str/replace "{uid}" (Filter/encodeValue ^String uid-string)))))
+
 (s/defn ^:private user-groups :- (s/maybe [su/NonBlankString])
   "Retrieve groups for a supplied DN."
-  [ldap-connection :- LDAPConnectionPool
-   dn              :- su/NonBlankString
-   {:keys [group-base]} :- i/LDAPSettings]
+  [ldap-connection                              :- LDAPConnectionPool
+   dn                                           :- su/NonBlankString
+   uid                                          :- su/NonBlankString
+   {:keys [group-base group-membership-filter]} :- i/LDAPSettings]
   (when group-base
     (let [results (ldap-client/search
                    ldap-connection
                    group-base
                    {:scope  :sub
-                    :filter (Filter/createEqualityFilter "member" ^String dn)})]
+                    :filter (process-group-membership-filter group-membership-filter dn uid)})]
       (map :dn results))))
 
 (s/defn ldap-search-result->user-info :- (s/maybe i/UserInfo)
   "Convert the result "
   [ldap-connection          :- LDAPConnectionPool
-   {:keys [dn], :as result} :- su/Map
+   {:keys [dn, uid], :as result} :- su/Map
```

Comment:
This is so trivial you shouldn't bother fixing it, but just a gentle reminder https://guide.clojure.style/#no-commas-for-seq-literals